pkg/test/utils/kubernetes.go

```diff
@@ -890,13 +890,7 @@ func StartTestEnvironment() (env TestEnvironment, err error) {
 		KubeAPIServerFlags: append(APIServerDefaultArgs, "--advertise-address={{ if .URL }}{{ .URL.Hostname }}{{ end }}"),
 	}
 
-	// Retry up to three times for the test environment to start up in case there is a port collision (#510).
-	for i := 0; i < 3; i++ {
-		env.Config, err = env.Environment.Start()
-		if err == nil {
-			break
-		}
-	}
+	env.Config, err = env.Environment.Start()
```

Comment:
I don't think this loop makes sense anymore. Envtest already has [logic to retry](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/envtest/server.go#L270) starting the controller plane.